libp2p/multiaddress.nim

```diff
@@ -909,12 +909,12 @@ proc append*(m1: var MultiAddress, m2: MultiAddress): MaResult[void] =
     ok()
 
 proc `&`*(m1, m2: MultiAddress): MultiAddress {.
-     raises: [Defect, ResultError[string]].} =
+     raises: [Defect, LPError].} =
   ## Concatenates two addresses ``m1`` and ``m2``, and returns result.
   ##
   ## This procedure performs validation of concatenated result and can raise
   ## exception on error.
-  concat(m1, m2).tryGet()
+  concat(m1, m2).expect("Invalid multiaddress after concatenation!")
```

Comment:
expect raises a defect - either the code or the comment is wrong

Reply:
Ah, I guess we can't really use `expect/get` with `toException`.

Reply:
A an observation, `raises: [Defect]` hides the fact that `expect/get` will raise a `Defect`. Not sure what's the best way to approach it is - maybe we shouldn't allow any exception to escape with `raises: []`?